api/operator/v1alpha1/vmdistributedcluster_types.go

```diff
@@ -0,0 +1,465 @@
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.updateStatus",description="current status of update rollout"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// VMDistributedClusterSpec is progressively rolling out updates to multiple VMClusters.
+type VMDistributedCluster struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is a standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty,omitzero"`
+
+	// spec defines the desired state of VMDistributedCluster
+	// +required
+	Spec VMDistributedClusterSpec `json:"spec"`
+
+	// status defines the observed state of VMDistributedCluster
+	// +optional
+	Status VMDistributedClusterStatus `json:"status,omitempty,omitzero"`
+
+	// ParsedLastAppliedSpec contains last-applied configuration spec
+	ParsedLastAppliedSpec *VMDistributedClusterSpec `json:"-" yaml:"-"`
+}
+
+// SelectorLabels defines selector labels for given component kind
+func (cr *VMDistributedCluster) SelectorLabels(kind vmv1beta1.ClusterComponent) map[string]string {
+	return vmv1beta1.ClusterSelectorLabels(kind, cr.Name, "vmd")
+}
+
+// PrefixedName returns prefixed name for the given component kind
+func (cr *VMDistributedCluster) PrefixedName(kind vmv1beta1.ClusterComponent) string {
+	return vmv1beta1.ClusterPrefixedName(kind, cr.Name, "vmd", false)
+}
+
+// FinalLabels adds cluster labels to the base labels and filters by prefix if needed
+func (cr *VMDistributedCluster) FinalLabels(kind vmv1beta1.ClusterComponent) map[string]string {
+	return vmv1beta1.AddClusterLabels(cr.SelectorLabels(kind), "vmd")
+}
+
+// AnnotationsFiltered returns global annotations to be applied by objects generate for vmcluster
+func (cr *VMDistributedCluster) AnnotationsFiltered() map[string]string {
+	return map[string]string{}
+}
+
+// AsOwner returns owner references with current object as owner
+func (cr *VMDistributedCluster) AsOwner() metav1.OwnerReference {
+	return metav1.OwnerReference{
+		APIVersion:         cr.APIVersion,
+		Kind:               cr.Kind,
+		Name:               cr.Name,
+		UID:                cr.UID,
+		Controller:         ptr.To(true),
+		BlockOwnerDeletion: ptr.To(true),
+	}
+}
+
+// PodLabels returns pod labels for given component kind
+func (cr *VMDistributedCluster) PodLabels(kind vmv1beta1.ClusterComponent) map[string]string {
+	selectorLabels := cr.SelectorLabels(kind)
+	podMetadata := cr.PodMetadata(kind)
+	if podMetadata == nil {
+		return selectorLabels
+	}
+	return labels.Merge(podMetadata.Labels, selectorLabels)
+}
+
+func (cr *VMDistributedCluster) GetVMAuthSpec() *vmv1beta1.VMAuthLoadBalancerSpec {
+	if cr == nil {
+		return nil
+	}
+	spec := cr.Spec.VMAuth.Spec
+	if spec == nil {
+		spec = &vmv1beta1.VMAuthLoadBalancerSpec{
+			CommonDefaultableParams: vmv1beta1.CommonDefaultableParams{
+				Port: "8417",
+				Image: vmv1beta1.Image{
+					Repository: "docker.io/victoriametrics/vmauth",
+					Tag:        "latest",
```

Comment:
it should be set to global operator version at `internal/config/config.go` also registry separately is expected to be configured